src/components/Simulation/SimulationBlocks.js

```diff
@@ -20,7 +20,16 @@ const SimulationBlocks = props => {
   // #TODO add some loading states
   const [error, setError] = useState(null);
   const [loading, setLoading] = useState(false);
-  const setsByCode = groupBy((data || {}).invoices, "code");
+  const invoices = (data || {}).invoices || [];
+
+  // Placeholder to only display the selected org unit, in the future
+  // the backend should do this filtering for us.
+  const displayedOrgUnit = props.searchQuery.orgUnit || "";
+  const invoicesForOrgUnit = invoices.filter(
```

Comment:
Please rebase from develop as optimized the filtering perf. This should be probably integrated within the memoized function, otherwise it will re-filter on each re-render. See https://github.com/BLSQ/hesabu-manager/commit/a1d80f29780b91cc8ccf1fc82cc66b364d94892e#diff-124b0b37de1543bcd7c6e2e1e2d9ca00R61